BUILDING.md

```diff
@@ -0,0 +1,124 @@
+# Building thin-edge.io
+
+## Installing toolchain
+
+### Rust toolchain
+
+To install Rust please follow [Official installation guide](https://www.rust-lang.org/tools/install).
+To get started you need Cargo's bin directory (`$HOME/.cargo/bin`) in your PATH
+environment variable.
+
+```shell
+export PATH=$PATH:$HOME/.cargo/bin
+```
+
+And then you can run `rustc` to view current version:
+
+```shell
+$ rustc --version
+rustc 1.51.0 (2fd73fabe 2021-03-23)
+```
+
+> Note: Above command will add rust to path only for existing session,
+> after you restart the session you will have to add it again,
+> to add rust to the path permanently it will depend on your shell but for Bash,
+> you simply need to add the line from above, `export PATH=$PATH:$HOME/.cargo/bin` to your ~/.bashrc.
+
+> For other shells, you'll want to find the appropriate place to set a configuration at start time,
+> eg. zsh uses ~/.zshrc. Check your shell's documentation to find what file it uses.
+
+thin-edge.io operates the `MSRV` (Minimum Supported Rust Version) and uses stable toolchain.
+
+Current MSRV is `1.46.0`.
+
+### Cross compilation toolchain
+
+thin-edge.io can be compiled for target architecture on non-target device, this is called cross compilation. Currently we support Raspberry Pi 3 building for `armv7` architecture with Rust's cross compilation toolchain called `cargo cross`.
```

Comment:
Raspberry Pi 3 => Raspberry Pi 3B. Actually, there is Raspberry Pi 3A, which I've never seen someone is using :D